File: app/src/main/java/in/hridayan/ashell/uix/ToastUtils.java
package in.hridayan.ashell.UI;

import android.widget.Toast;
import android.content.Context;

public class ToastUtils {
  private static Toast toast;
  public static int LENGTH_SHORT = 0, LENGTH_LONG = 1;

  public static void showToast(Context context, String message, int length) {
    if (toast != null) {
      toast.cancel();
    }
    toast = Toast.makeText(context, message, length);
    toast.show();
  }

  public static void showToast(Context context, int resId, int length) {
    showToast(context, context.getString(resId), length);
  }
}

File: app/src/main/java/in/hridayan/ashell/ui/ToastUtils.kt
package `in`.hridayan.ashell.ui

import android.content.Context
import android.widget.Toast

object ToastUtils {
    private var toast: Toast? = null
    @JvmField
    var LENGTH_SHORT: Int = 0
    @JvmField
    var LENGTH_LONG: Int = 1

    @JvmStatic
    fun showToast(context: Context?, message: String?, length: Int) {
        if (toast != null) {
            toast?.cancel()
        }
        toast = Toast.makeText(context, message, length)
        toast?.show()
    }

    @JvmStatic
    fun showToast(context: Context, resId: Int, length: Int) {
        showToast(context, context.getString(resId), length)
    }
}
